### PR TITLE
Migrate API tests to GitHub workflow

### PR DESCRIPTION
**Pull request by nsoranzo**

### CODE REVIEW

File: .github/workflows/api.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip dir
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: pip-cache-${{ matrix.python-version }}-${{ hashFiles('galaxy root/requirements.txt') }}
+      - name: Run tests
+        run: ./run_tests.sh --skip_flakey_fails -api
+        working-directory: 'galaxy root'
```
Comment (mvdbeek) [inline .github/workflows/api.yaml:35]: Can we publish `run_api_tests.html` ?

**Comment by mvdbeek** [comment]
```
2020-11-10T14:51:22.9522306Z galaxy.tool_util.output_checker INFO 2020-11-10 13:37:01,679 Job failed because of contents in the standard error stream: [python: can't open file '/home/runner/work/galaxy/galaxy/galaxy': [Errno 2] No such file or directory
```
Looks like we have some test tools using unquoted paths ?

**Comment by mvdbeek** [comment]
https://github.com/galaxyproject/galaxy/blob/dev/tools/filters/catWrapper.xml#L3 ...

File: tools/filters/catWrapper.xml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
     <description>tail-to-head</description>
-    <command interpreter="python">
-        catWrapper.py
-        $out_file1
-        $input1
-        #for $q in $queries
-            ${q.input2}
-        #end for
-    </command>
+    <command interpreter="python"><![CDATA[
```
Comment (mvdbeek) [inline tools/filters/catWrapper.xml:3]: I think the command interpreter part doesn't quote the path.

Comment (nsoranzo) [reply tools/filters/catWrapper.xml:3]: Yep, fixed that in the next commit!

**Comment by nsoranzo** [comment]
Tests finally pass (Integration Selenium failure is not related).

File: lib/galaxy/tools/evaluation.py [hunk 3]
```diff
@@ -494,11 +495,12 @@ def __build_command_line(self):
...
             # TODO: path munging for cluster/dataset server relocatability
-            executable = command_line.split()[0]
+            command_line_tokens = shlex.split(command_line)
+            executable = command_line_tokens[0]
             tool_dir = os.path.abspath(self.tool.tool_dir)
             abs_executable = os.path.join(tool_dir, executable)
-            command_line = command_line.replace(executable, abs_executable, 1)
-            command_line = interpreter + " " + command_line
+            command_line_tokens[0:1] = [interpreter, abs_executable]
+            command_line = ' '.join(map(shlex.quote, command_line_tokens))
```
Comment (jmchilton) [inline lib/galaxy/tools/evaluation.py:503]: Can we revert this commit - it breaks tools that use environment variables (?)


```
python2 /mnt/galaxy/data/galaxy/galaxy-toolshed-tools/testtoolshed.g2.bx.psu.edu/repos/peterjc/mira4_9_mirabait/5979f13c2dc5/mira4_9_mirabait/tools/mira4_9/mirabait/./mira_check_version.py '${MIRA4_9}mirabait' 4.9 '&&' '${MIRA4_9}mirabait' -k 32 -n 6 -b
```

is very different than

```
python2 /mnt/galaxy/data/galaxy/galaxy-toolshed-tools/testtoolshed.g2.bx.psu.edu/repos/peterjc/mira4_9_mirabait/5979f13c2dc5/mira4_9_mirabait/tools/mira4_9/mirabait/./mira_check_version.py ${MIRA4_9}mirabait 4.9 && ${MIRA4_9}mirabait -k "25" -n "6" -b
```

shlex quoting all the arguments in this one scenario and not in all of them also makes it hard to track down these problems and leads to fairly inconsistent behavior right?

Comment (nsoranzo) [reply lib/galaxy/tools/evaluation.py:503]: https://github.com/galaxyproject/galaxy/pull/11343

**Pull request merged**